File: src/main/java/jenkem/shared/data/ImageHtml.java
package jenkem.shared.data;

import java.io.Serializable;
import javax.jdo.annotations.IdentityType;
import javax.jdo.annotations.PersistenceCapable;
import javax.jdo.annotations.PrimaryKey;

/**
 * Persistent HTML data for converted images.
 */
@PersistenceCapable(identityType = IdentityType.APPLICATION)
public class ImageHtml implements ImagePartIfc, Serializable {
    private static final long serialVersionUID = 7068931434659156682L;

    @PrimaryKey
    private String _id; // mongodb identifier
    private String name;
    private String html;

    public ImageHtml() {
    }

    public ImageHtml(final String name, final String html) {
        this._id = name;
        this.name = name;
        this.html = html;
    }

    public final String getName() {
        return name;
    }

    public final String getHtml() {
        return html;
    }

    @Override
    public final String toString() {
        return name;
    }
}

File: src/main/scala/jenkem/persistence/data/ImageHtml.scala
package jenkem.persistence.data

import javax.jdo.annotations.IdentityType
import javax.jdo.annotations.PersistenceCapable
import javax.jdo.annotations.PrimaryKey

/**
 * Persistent HTML data import jenkem.persistence.data.ImagePartTrait
for converted images.
 */
@PersistenceCapable(identityType = IdentityType.APPLICATION)
class ImageHtml(var name: String, var html: String) extends ImagePartTrait {
  @PrimaryKey var _id = name
  override def toString: String = name
}
